### PR TITLE
Fixing dashboard tests

### PR DESCRIPTION
**Pull request by lickem22**
Reviewer: @sidravi1 
Estimate: @20mins

---

## Ticket

Fixes: HOTFIX

### Goal
The goal of this PR is to improve dashboards so that it does not only check for the length of the array returned, but for the content as well 
## How has this been tested?
 Ran the tests

## Checklist

Fill with `x` for completed. 

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts

(Delete any items below that are not relevant)
- [x] I have updated the automated tests

### CODE REVIEW

File: core_backend/tests/api/test_question_dashboard.py [hunk 1]
```diff
@@ -23,3 +52,7 @@ async def test_retrieve_question_dashboard(
     data = response.json()
     assert len(data["six_months_questions"]) == 6
     assert len(data["six_months_upvotes"]) == 6
+    assert all(value >= questions[0] for value in data["six_months_questions"])
```
Comment (lickem22) [inline core_backend/tests/api/test_question_dashboard.py:55]: Instead of checking each values individually I make sure the values are within the treshold?

File: core_backend/tests/api/test_question_dashboard.py [hunk 2]
```diff
@@ -1,17 +1,46 @@
...
+async def mock_dashboard_stats(
+    user_id: int, asession: None, questions: tuple, upvotes: tuple
+) -> QuestionDashBoard:
+    """
+    Function to monkeypatch question_dashboard.routersaz.get_dashboard_stats with
+    random data
+    """
+    return QuestionDashBoard(
+        six_months_questions=[
+            random.randint(questions[0], questions[1]) for _ in range(6)
```
Comment (sidravi1) [inline core_backend/tests/api/test_question_dashboard.py:19]: My goal was to check for corner cases. I see that you are doing that by including (0, 0) but this may just make it more complex (took me a while to understand it) and I'm unsure of the benefit. 

Can we set a seed to make this deterministic / replicable?

File: core_backend/tests/api/test_question_dashboard.py (hunk 1 above)
Comment (sidravi1) [reply core_backend/tests/api/test_question_dashboard.py:55]: My vote is to just make it fixed values. See comment above!

**Review by sidravi1 — APPROVED** [review]
Thanks @lickem22!

Suggest just making the test simpler by using fixed values.

File: core_backend/tests/api/test_question_dashboard.py (hunk 2 above)
Comment (lickem22) [reply core_backend/tests/api/test_question_dashboard.py:19]: I did it that way because it is just too much static code to do it that way, because you need to enter 6 values twice for every case and then check it one by one. Is there any value to make it replicable?

**Comment by sidravi1** [comment]
Just nice to have deterministic tests so if it passes locally we are
confident it will pass in CI/CD. Please add a seed if you want to keep it
the current way
